lib/src/rules/prefer_int_literals.dart

```diff
@@ -71,26 +71,26 @@ class _Visitor extends SimpleAstVisitor<void> {
     if (node is FunctionExpression) {
       var functDeclaration = node.parent;
       if (functDeclaration is FunctionDeclaration) {
-        return functDeclaration.returnType?.type?.name == 'double';
+        return functDeclaration.returnType?.type?.isDartCoreDouble == true;
       }
     } else if (node is MethodDeclaration) {
-      return node.returnType?.type?.name == 'double';
+      return node.returnType?.type?.isDartCoreDouble == true;
     }
     return false;
   }
 
   bool hasTypeDouble(Expression expression) {
     final parent = expression.parent;
     if (parent is ArgumentList) {
-      return expression.staticParameterElement?.type?.name == 'double';
+      return expression.staticParameterElement?.type?.isDartCoreDouble == true;
     } else if (parent is ListLiteral) {
       final typeArguments = parent.typeArguments?.arguments;
       return typeArguments?.length == 1 &&
-          typeArguments[0]?.type?.name == 'double';
+          typeArguments[0]?.type?.isDartCoreDouble == true;
```

Comment:
`.type` of a type arguments cannot be null, so could be just `typeArguments[0].type.isDartCoreDouble`

Reply:
👍